src/services/game-logic/GameEngine.ts

```diff
@@ -30,6 +30,22 @@ class GameEngine {
         }
     }
 
+    public changePawnAfterPromotion(square: Square, selectedPiece: PieceNames, color: Colors): void {
+        this.board.promotePawn(square, selectedPiece, color);
+    }
+
+    public checkPawnPromotion(destination: Square): Colors | null {
+        const piece = this.board.getPiece(destination);
+
+        if (piece?.name === PieceNames.PAWN && destination.row === 0 && piece?.color === Colors.WHITE) {
+            return Colors.WHITE;
+        }
+        if (piece?.name === PieceNames.PAWN && destination.row === 7 && piece?.color === Colors.BLACK) {
```

Comment:
ja z code review się nauczyłam, żeby takie cyferki choć zwięzłe i wygodne lepiej zamieniać zmiennymi. Jakby co to w enumach zrobiłam plik constants na potrzeby roszady, więc może Ci się przyda, żeby tam wprowadzić nazwy na rzędy